File: Module.Frontend.TPM/Client/model/tpm/product/HistoricalProduct.js
Ext.define('App.model.tpm.product.HistoricalProduct', {
    extend: 'Ext.data.Model',
    idProperty: '_Id',
    breezeEntityType: 'Product',
    fields: [
         { name: '_Id', type: 'string', hidden: true },
        { name: '_ObjectId', hidden: true },
        { name: '_User', type: 'string', isDefault: true },
        { name: '_Role', type: 'string', isDefault: true },
        { name: '_EditDate', type: 'date', isDefault: true },
        { name: '_Operation', type: 'string', isDefault: true },
        { name: 'ZREP', useNull: true, type: 'string', hidden: false, isDefault: true },
        { name: 'EAN', useNull: true, type: 'string', hidden: false, isDefault: true },
        { name: 'ProductRU', useNull: true, type: 'string', hidden: false, isDefault: true },
        { name: 'ProductEN', useNull: true, type: 'string', hidden: false, isDefault: true },
        { name: 'BrandFlagAbbr', useNull: true, type: 'string', hidden: false, isDefault: true },
        { name: 'BrandFlag', useNull: true, type: 'string', hidden: false, isDefault: true },
        { name: 'SubmarkFlag', useNull: true, type: 'string', hidden: false, isDefault: true },
        { name: 'IngredientVariety', useNull: true, type: 'string', hidden: false, isDefault: true },
        { name: 'ProductCategory', useNull: true, type: 'string', hidden: false, isDefault: true },
        { name: 'ProductType', useNull: true, type: 'string', hidden: false, isDefault: true },
        { name: 'MarketSegment', useNull: true, type: 'string', hidden: false, isDefault: true },
        { name: 'SupplySegment', useNull: true, type: 'string', hidden: false, isDefault: true },
        { name: 'FunctionalVariety', useNull: true, type: 'string', hidden: false, isDefault: true },
        { name: 'Size', useNull: true, type: 'string', hidden: false, isDefault: true },
        { name: 'BrandEssence', useNull: true, type: 'string', hidden: false, isDefault: true },
        { name: 'PackType', useNull: true, type: 'string', hidden: false, isDefault: true },
        { name: 'GroupSize', useNull: true, type: 'string', hidden: false, isDefault: true },
        { name: 'TradedUnitFormat', useNull: true, type: 'string', hidden: false, isDefault: true },
        { name: 'ConsumerPackFormat', useNull: true, type: 'string', hidden: false, isDefault: true },
        { name: 'UOM_PC2Case', useNull: true, type: 'int', hidden: false, isDefault: true }
    ],
    proxy: {
        type: 'breeze',
        resourceName: 'HistoricalProducts',
        reader: {
            type: 'json',
            totalProperty: 'inlineCount',
            root: 'results'
        }
    }
});

File: Module.Persist.TPM/Model/History/HistoricalProduct.cs
using Core.History;
using Module.Persist.TPM.Model.TPM;
using System;

namespace Module.Persist.TPM.Model.History
{
    [AssociatedWith(typeof(Product))]
    public class HistoricalProduct : BaseHistoricalEntity<System.Guid>
    {
        public string ZREP { get; set; }
        public string EAN_Case { get; set; }
        public string EAN_PC { get; set; }
        public string ProductEN { get; set; }
		public string Brand { get; set; }
		public string Brand_code { get; set; }
		public string Technology { get; set; }
		public string Tech_code { get; set; }
		public string BrandTech { get; set; }
		public string BrandTech_code { get; set; }
		public string Segmen_code { get; set; }
		public string BrandsegTech_code { get; set; }
        public string Brandsegtech{ get; set; }
        public string BrandsegTechsub_code { get; set; }
        public string BrandsegTechsub { get; set; }
        public string SubBrand_code { get; set; }
        public string SubBrand { get; set; }
        public string BrandFlagAbbr { get; set; }
        public string BrandFlag { get; set; }
        public string SubmarkFlag { get; set; }
        public string IngredientVariety { get; set; }
        public string ProductCategory { get; set; }
        public string ProductType { get; set; }
        public string MarketSegment { get; set; }
        public string SupplySegment { get; set; }
        public string FunctionalVariety { get; set; }
        public string Size { get; set; }
        public string BrandEssence { get; set; }
        public string PackType { get; set; }
        public string GroupSize { get; set; }
        public string TradedUnitFormat { get; set; }
        public string ConsumerPackFormat { get; set; }

        public int? UOM_PC2Case { get; set; }
		public int? Division { get; set; }
	}
}